pdr_backend/pdr_dashboard/callbacks/callbacks_feeds.py

```diff
@@ -0,0 +1,254 @@
+        for key in ["addr", "base_token", "quote_token"]
+    )
+
+
+def range_condition(item, field, min_value, max_value):
+    item_value = float(item[field])
+    return not (
+        (min_value is not None and min_value != "" and item_value < min_value)
+        or (max_value is not None and max_value != "" and item_value > max_value)
+    )
+
+
+def check_condition(item, condition_type, field, *values):
+    if condition_type == "search":
+        return search_condition(item, values[0])
+    if condition_type == "filter":
+        return filter_condition(item, field, values[0])
+    if condition_type == "range":
+        return range_condition(
+            item,
+            field,
+            float(values[0]) if values[0] is not None and values[0] != "" else None,
+            float(values[1]) if values[1] is not None and values[1] != "" else None,
+        )
+    return True
+
+
+def get_callbacks_feeds(app):
+    @app.callback(
+        Output("feeds_page_table", "data"),
+        [
+            Input("base_token", "value"),
+            Input("quote_token", "value"),
+            Input("exchange", "value"),
+            Input("time", "value"),
+            Input("sales_button", "n_clicks"),
+            Input("revenue_button", "n_clicks"),
+            Input("accuracy_button", "n_clicks"),
+            Input("volume_button", "n_clicks"),
+            Input("search-input-feeds-table", "value"),
+        ],
+        State("sales_min", "value"),
+        State("sales_max", "value"),
+        State("revenue_min", "value"),
+        State("revenue_max", "value"),
+        State("accuracy_min", "value"),
+        State("accuracy_max", "value"),
+        State("volume_min", "value"),
+        State("volume_max", "value"),
+        prevent_initial_call=True,
+    )
+    def filter_table(
+        base_token,
+        quote_token,
+        exchange,
+        time,
+        _n_clicks_sales,
+        _n_clicks_revenue,
+        _n_clicks_accuracy,
+        _n_clicks_volume,
+        search_input_value,
+        sales_min,
+        sales_max,
+        revenue_min,
+        revenue_max,
+        accuracy_min,
+        accuracy_max,
+        volume_min,
+        volume_max,
+    ):
+        """
+        Filter table based on selected dropdown values.
+        """
+
+        conditions = [
+            ("filter", "base_token", base_token),
+            ("filter", "quote_token", quote_token),
+            ("filter", "exchange", exchange),
+            ("filter", "time", time),
+            ("range", "sales_raw", sales_min, sales_max),
+            ("range", "sales_revenue_(OCEAN)", revenue_min, revenue_max),
+            ("range", "avg_accuracy", accuracy_min, accuracy_max),
+            ("range", "volume_(OCEAN)", volume_min, volume_max),
+        ]
+
+        new_table_data = [
+            item
+            for item in app.feeds_table_data
+            if (
+                all(check_condition(item, *condition) for condition in conditions)
```

Comment:
you can initialize the conditions first, then check them for every item. Otherwise this is recomputed in the loop every time.